packages/shared/src/components/post/PostContent.tsx

```diff
@@ -96,6 +103,17 @@ export function PostContent({
     onRemovePost,
   };
 
+  const {
+    onPostArticleClick: onReadArticleClick,
+    activateCompanion,
+    openArticle,
+    isOpen: companionModalOpen,
+    toggleOpen: companionModalToggle,
+
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+  } = useCompanionTrigger(onReadArticle);
```

Comment:
Yeah seeing this, by converting to lazyModal we should be able to move all logic to one place :)